src/PokemonBase.java

```diff
@@ -0,0 +1,33 @@
+public abstract class PokemonBase {
+
+    private String name;
+    protected enum Type {ELECTRIC, FIRE, GRASS, WATER};
+    private int hp;
+    private Type type;
+
+    public PokemonBase(String name, Type type) {
+        this.name = name;
+        this.type = type;
+        this.hp = 100;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void decreaseHp(int points) {
+        hp -= points;
+        if (hp < 0) {
+            hp = 0;
+        }
+    }
+    public int getHp() {
+        return hp;
+    }
+
+    public String toString() {
```

Comment:
Dit is een hele handige toevoeging om de klasse te kunnen printen. Vergeet niet om hier ook `@Override` boven te zetten voor de duidelijkheid. Dit overschrijf je namelijk vanuit de `Object` klasse. Alles is een `Object` in java (behalve primitieven)